validator/client/beacon-api/beacon_api_validator_client.go

```diff
@@ -14,24 +14,30 @@ import (
 )
 
 type beaconApiValidatorClient struct {
-	url            string
-	httpClient     http.Client
-	fallbackClient iface.ValidatorClient
+	url                string
+	httpClient         http.Client
+	fallbackClient     iface.ValidatorClient
+	genesisProvider    genesisProvider
+	domainDataProvider domainDataProvider
```

Comment:
I would not create a separate `domainDataProvider` type. See https://github.com/prysmaticlabs/prysm/pull/11714#discussion_r1038382281